### PR TITLE
⬆️ upgrades dependencies

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

This batch of upgrades focused on brining FastAPI to the latest version. The following packages were upgraded:
- [x] `httpx==0.21.3`
- [x] `respx==0.19.1` (released constraint on respx~=0.17.1)
- [x] `fastapi==0.71.0` only minor adjustments to tests due to how starlette's `TestClient` works. Note that FastAPI has added and removed new packages since the previous release. 

Other libs:
- [x] `black==21.12b0`
- [x] `aiofiles==0.8.0`

## Related issue/s

- https://github.com/ITISFoundation/osparc-issues/issues/428

### CODE REVIEW

File: services/director-v2/tests/integration/02/utils.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ async def assert_retrieve_service(
         f"/v2/dynamic_services/{service_uuid}:retrieve",
         json=dict(port_keys=[]),
         headers=headers,
-        allow_redirects=True,
+        follow_redirects=True,
```
Comment (pcrespov) [inline services/director-v2/tests/integration/02/utils.py:233]: is this an options renamed in the new a API or a fix?

File: services/director-v2/tests/unit/with_dbs/test_route_clusters.py [hunk 2]
```diff
@@ -108,19 +109,21 @@ def creator(**overrides) -> Cluster:
...
 
 
-def test_get_default_cluster_entrypoint(clusters_config: None, client: TestClient):
+async def test_get_default_cluster_entrypoint(
+    loop: AbstractEventLoop, clusters_config: None, async_client: httpx.AsyncClient
+):
     # This test checks that the default cluster is accessible
     # the default cluster is the osparc internal cluster available through a dask-scheduler
-    response = client.get("/v2/clusters/default")
+    response = await async_client.get("/v2/clusters/default")
```
Comment (pcrespov) [inline services/director-v2/tests/unit/with_dbs/test_route_clusters.py:117]: Why did you switch to an asyn client i.e. what different does it make to the test?

**Review by pcrespov — APPROVED** [review]
Great! We had this constraint for a while! Added some questions

I also suggest to keep in sync the versions of ``black`` in the ``.pre-commit-config.yaml``.  It would be very annoying if vscode tools (using the version in ``_tools.txt``) behaves in one way
and when you commit everything gets re-formatted in another due to the commit hooks :-)

So, all in all: we should not forget to keep the pre-commit tools versions in sync with those used locally (they are installed in different venv)

File: services/director-v2/tests/integration/02/utils.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/tests/integration/02/utils.py:233]: the API changed here, had to adapt the code as well.

File: services/director-v2/tests/unit/with_dbs/test_route_clusters.py (hunk 2 above)
Comment (GitHK) [reply services/director-v2/tests/unit/with_dbs/test_route_clusters.py:117]: I've already put a comment in a fixture but basically Starlette's TestClient creates a loop in a different thread, causing issues. Using the async client all code is attached to the same loop.

**Review by sanderegg — APPROVED** [review]
great!

**Pull request merged**